parachain/pallets/bridge/src/lib.rs

```diff
@@ -85,54 +104,26 @@ decl_module! {
 		fn deposit_event() = default;
 
 		#[weight = 0]
-		pub fn submit(origin, channel_id: ChannelId, app_id: AppId, message: Message) -> DispatchResult {
+		pub fn submit(origin, message: Message) -> DispatchResult {
 			let relayer = ensure_signed(origin)?;
 
-			let mut channel = make_inbound_channel::<T>(channel_id);
-			channel.submit(&relayer, app_id, &message)
-		}
-
-		/// Submit multiple messages for dispatch to multiple target applications.
-		#[weight = 0]
-		pub fn submit_bulk(origin, messages: Vec<(AppId, Message)>) -> DispatchResult {
-			let who = ensure_signed(origin)?;
-
-			debug::RuntimeLogger::init();
-
-			debug::trace!(
-				target: "submit_messages",
-				"Received {} messages",
-				messages.len(),
-			);
-
-			T::Verifier::verify_bulk(who, messages.as_slice())?;
-
-			debug::trace!(
-				target: "submit_messages",
-				"Message verification succeeded",
-			);
-
-			let errors: Vec<DispatchError> = messages.iter()
-				.map(|(app_id, msg)| Self::dispatch(app_id.into(), msg))
-				.filter_map(|r| r.err())
-				.collect();
-
-			debug::trace!(
-				target: "submit_messages",
-				"Messages were dispatched",
-			);
+			let log = T::Verifier::verify(&message)?;
+			let envelope = Envelope::try_from(log).map_err(|_| Error::<T>::InvalidEnvelope)?;
+			let channel_id = SourceChannels::get(envelope.channel)
+				.ok_or(Error::<T>::InvalidSourceChannel)?;
+			let channel = make_inbound_channel::<T>(channel_id);
 
-			Ok(())
+			channel.submit(&relayer, &envelope)
 		}
 	}
 }
 
 impl<T: Config> Module<T> {
-	fn dispatch(address: H160, message: &Message) -> DispatchResult {
-		if address == T::AppETH::address() {
-			T::AppETH::handle(message.payload.as_ref())
-		} else if address == T::AppERC20::address() {
-			T::AppERC20::handle(message.payload.as_ref())
+	fn dispatch(source: H160, payload: &[u8]) -> DispatchResult {
+		if source == T::AppETH::address() {
```

Comment:
i see the source->destination mapping from eth to substrate is now hardcoded/defined here, rather than being based on something like a pallet_id - is this correct? if so, sounds good for now and we can evolve it later.

Reply:
Yeah we definitely need to resolve that by encoding the `(PalletId, CallIndex)` in the solidity contract. Its gonna make this much cleaner.

I have a plan for doing it and should be straightforward, after seeing how they did it in parity-bridges-common.

With this change, the ETH pallet will have this dispatchable: 
```
fn mint(origin: Origin, recipient, amount) {
  sender_contract = ensure_signed!(origin);
  ensure!(sender_contract == AccountIdForETHApp)
  ...
}
```

where `AccountIdForETHApp` is a special AccountId that doesn't have a private key and is derived from the H160 address of the ETHApp contract.

Reply:
cool leave it for a follow up PR though :P

Reply:
Added #239 to track this.